### PR TITLE
feat: adding admin session listing api

### PR DESCRIPTION
**Pull request by kelkarajay**
## Changes

- Adds admin session listing API

## Related issue(s)

Partially resolves - [Console: View & terminate sessions](https://github.com/ory-corp/cloud/issues/2007)

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: session/handler.go [hunk 1]
```diff
@@ -261,6 +262,61 @@ func (h *Handler) adminDeleteIdentitySessions(w http.ResponseWriter, r *http.Req
...
+// swagger:parameters adminListSessions
+// nolint:deadcode,unused
+type adminListSessions struct {
+	// Active is a boolean flag that filters out sessions based on the state. If no value is provided, all sessions are returned.
+	//
+	// required: false
+	// in: query
+	Active bool `json:"active"`
+
+	x.PaginationParams
```
Comment (aeneasr) [inline session/handler.go:274]: Please follow our new guidelines around pagination :)

https://intranet.ory.sh/docs/dev/api-guidelines#pagination

**Review by aeneasr — COMMENTED** [review]
Looks pretty good already! :)

**Comment by kelkarajay** [comment]
Requires - https://github.com/ory/x/pull/619

File: session/handler.go [hunk 2]
```diff
@@ -263,6 +269,115 @@ func (h *Handler) adminDeleteIdentitySessions(w http.ResponseWriter, r *http.Req
...
+	//
+	// This is the number of items per page to return.
+	// For details on pagination please head over to the [pagination documentation](https://www.ory.sh/docs/ecosystem/api-design#pagination).
+	//
+	// required: false
+	// in: query
+	// default: 250
+	// min: 1
+	// max: 1000
+	PageSize int `json:"page_size"`
```
Comment (aeneasr) [inline session/handler.go:289]: Use structs from ory/x here?

File: session/handler.go [hunk 3]
```diff
@@ -263,6 +269,115 @@ func (h *Handler) adminDeleteIdentitySessions(w http.ResponseWriter, r *http.Req
...
+	opts, err := keysetpagination.Parse(&urlValues)
+	if err != nil {
+		h.r.Writer().WriteError(w, r, herodot.ErrBadRequest.WithError("could not parse parameter page_size"))
+		return
+	}
+
+	opts = append(opts, keysetpagination.WithDefaultSize(paginationDefaultItems))
+	opts = append(opts, keysetpagination.WithMaxSize(paginationMaxItems))
+
+	sess, total, nextPage, err := h.r.SessionPersister().ListSessions(r.Context(), active, keysetpagination.GetPaginator(opts...), ExpandEverything)
```
Comment (aeneasr) [inline session/handler.go:370]: Maybe better to not expand by default?

**Review by aeneasr — COMMENTED** [review]
Looks really nice already!

File: session/handler.go (hunk 3 above)
Comment (kelkarajay) [reply session/handler.go:370]: True. Should probably also make this a parameter like the `active` filter option

File: session/handler.go (hunk 2 above)
Comment (kelkarajay) [reply session/handler.go:289]: yep. [will basically use this](https://github.com/ory/x/pull/619)

Comment (kelkarajay) [reply session/handler.go]: Fixed to use `keysetpagination.RequestParameters`

File: session/test/persistence.go [hunk 4]
```diff
@@ -196,13 +198,108 @@ func TestPersister(ctx context.Context, conf *config.Config, p interface {
...
+						expected: []session.Session{
+							sess[1],
+							sess[3],
+						},
+					},
+				} {
+					t.Run("case=ListSessions "+tc.desc, func(t *testing.T) {
+						paginatorOpts := make([]keysetpagination.Option, 0)
+						paginatorOpts = append(paginatorOpts, keysetpagination.WithDefaultSize(250))
+						paginatorOpts = append(paginatorOpts, keysetpagination.WithMaxSize(1000))
```
Comment (aeneasr) [inline session/test/persistence.go:240]: Shall we test with a token as well? So `0000-0000...` as well as a proper token?

Comment (kelkarajay) [reply session/test/persistence.go]: test added that iterates over pages

**Review by jonas-jonas — APPROVED** [review]
LGTM.

**Review by aeneasr — DISMISSED** [review]
Nice! LGTM!

Linter is complaining with a minor thing, but this is mergable otherwise!

**Comment by kelkarajay** [comment]
@jonas-jonas @aeneasr the pagination fix for mysql is now included with the ory/x version on this branch. re-review please?

**Review by aeneasr — APPROVED** [review]
👌

**Pull request merged**